exonum-java-binding/common/src/main/java/com/exonum/binding/common/message/SignedMessage.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.message;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.messages.Consensus;
+import com.exonum.binding.messages.Consensus.ExonumMessage;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * A wrapper around {@link Consensus.SignedMessage} protobuf message containing
```

Comment:
It's not clear what is the purpose of this class and where it can be helpful. As I can see in `PassedMessage` we can get everything we need from the protobuf message.

Reply:
🤷‍♂ The line below says " which converts protobuf types into internal types." That is its main purpose.

You do need such a class, unless you are fine with such client code:
```
PublicKey authorPk = PublicKey.fromBytes(message.getAuthor()
        .toByteArray());
```

Can you spot a mistake in this snippet — it compiles, will not throw in runtime, but is incorrect?

Reply:
On top of that — it stores ExonumMessage as ByteString, which is even less intuitive to use.